### PR TITLE
Fix daml-lf-ast build on Windows

### PR DESCRIPTION
**Pull request by nmattia**
This contains a few changes described below. They are bundled to reduce CI contention and because they are related. The end result is a successful build of daml-lf-ast on Windows.

### Update rules_haskell

This updates to the latest rules_haskell, which fixes a few issues on
Windows. Most importantly it flags a few Windows libraries as "system"
libraries, preventing Hazel to fail because they are not provided
through Bazel.

### Fix the streaming-commons build on Windows

This modifies our custom streaming-commons BUILD file to make it work on
Windows. In particular it swaps some system libraries, passes the
`-DWINDOWS` flag and enables the build of an extra module.

### Fix shared object issues on Windows

* This fixes rules_haskell to use the correct Windows path separator on Windows.
    GHC expects the LD_LIBRARY_PATH variable to be a list of semi-colon separated
    paths, as opposed to a list of colon separated paths:
    https://github.com/ghc/ghc/blob/51fd357119b357c52e990ccce9059c423cc49406/compiler/ghci/Linker.hs#L1646-L1650

* This fixes the name of Haskell shared objects on Windows. By default
    Bazel's cc_library generates '.so' files, whereas GHC expects a `.dll`
    (or a few other extensions, non of which are `.so`):
    https://github.com/ghc/ghc/blob/51fd357119b357c52e990ccce9059c423cc49406/rts/linker/PEi386.c#L684

### Clean up bazel_tools BUILD file

This wraps a very long line for legibility.

### CODE REVIEW

File: bazel_tools/haskell-win-so-name.patch [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+     "is_shared_library",
+     "make_path",
+ )
+@@ -144,6 +145,13 @@ def get_libs_for_ghc_linker(hs, transitive_cc_dependencies, path_prefix = None):
+ 
+         # Additionally ghc 8.4 requires library_deps here although 8.6 does not
+         ld_library_deps = library_deps + _ld_library_deps
++
++    if hs.toolchain.is_windows:
++        # GHC's builtin linker requires .dylib files on MacOS.
```
Comment (moritzkiefer-da) [inline bazel_tools/haskell-win-so-name.patch]: Looks like you forgot to update this comment.

Comment (nmattia-da) [reply bazel_tools/haskell-win-so-name.patch]: Thanks.

**Comment by nmattia-da** [comment]
Fixed the issue on macOS, it was a matter of `if` versus `elif`.

**Comment by martin-drhu-da** [comment]
This is great progress. Thanks a lot.
`daml-lf-ast` used to depend on the DAML-LF protobuf description but it doesn't anymore. Thus, `daml-lf-proto` would be another very interesting target to port.

**Pull request merged**